frontend/src/pages/pipelines/global/pipelines/PipelinesPageHeaderActions.tsx

```diff
@@ -1,39 +1,76 @@
 import * as React from 'react';
-import { Dropdown, DropdownItem, DropdownSeparator, DropdownToggle } from '@patternfly/react-core';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownSeparator,
+  MenuToggle,
+  Tooltip,
+} from '@patternfly/react-core';
 import { DeleteServerModal, ViewServerModal } from '~/concepts/pipelines/context';
 
-const PipelinesPageHeaderActions: React.FC = () => {
+type PipelinesPageHeaderActionsProps = {
+  toggleElement: React.ReactNode;
+  variant: 'plain' | 'secondary';
+  isDisabled: boolean;
+};
+
+const PipelinesPageHeaderActions: React.FC<PipelinesPageHeaderActionsProps> = ({
+  toggleElement,
+  variant,
+  isDisabled,
+}) => {
   const [open, setOpen] = React.useState(false);
   const [deleteOpen, setDeleteOpen] = React.useState(false);
   const [viewOpen, setViewOpen] = React.useState(false);
+  const onToggleClick = () => {
+    setOpen(!open);
+  };
+
+  const DropdownComponent = (
+    <Dropdown
+      removeFindDomNode
+      onSelect={() => setOpen(false)}
+      toggle={
+        <MenuToggle
+          aria-label="pipeline-server-actions"
+          isDisabled={isDisabled}
+          variant={variant}
+          onClick={onToggleClick}
+        >
+          {toggleElement}
+        </MenuToggle>
+      }
+      isOpen={open}
+      position="right"
+      dropdownItems={[
+        <DropdownItem key="view-server-details" onClick={() => setViewOpen(true)}>
+          View pipeline server configuration
+        </DropdownItem>,
+        <DropdownSeparator key="separator" />,
+        <DropdownItem
+          onClick={() => {
+            setDeleteOpen(true);
+          }}
+          key="delete-server"
+        >
+          Delete pipeline server
+        </DropdownItem>,
+      ]}
+    />
+  );
 
   return (
     <>
-      <Dropdown
-        removeFindDomNode
-        onSelect={() => setOpen(false)}
-        toggle={
-          <DropdownToggle toggleVariant="secondary" onToggle={() => setOpen(!open)}>
-            Pipeline server actions
-          </DropdownToggle>
-        }
-        isOpen={open}
-        position="right"
-        dropdownItems={[
-          <DropdownItem key="view-server-details" onClick={() => setViewOpen(true)}>
-            View pipeline server configuration
-          </DropdownItem>,
-          <DropdownSeparator key="separator" />,
-          <DropdownItem
-            onClick={() => {
-              setDeleteOpen(true);
-            }}
-            key="delete-server"
-          >
-            Delete pipeline server
-          </DropdownItem>,
-        ]}
-      />
+      {isDisabled ? (
+        <Tooltip
+          content="To access the pipeline server actions, first create a pipeline server."
+          position="bottom"
+        >
+          {DropdownComponent}
+        </Tooltip>
+      ) : (
+        { DropdownComponent }
+      )}
```

Comment:
I had thought of modifying the z-index of the tooltip to hide it when `isDisabled` is `false` coz there is no tooltip content when `isDisabled` is `false`. Not sure if that's better.

Reply:
Why not open the tooltip to the right and omit it altogether for the action which has a text button?
cc @kywalker-rh

Reply:
Done. I've removed the tooltip for the text button(both disabled and enabled states). I'm currently only showing the tooltip(for disabled state) for the kebab menu. I'm not showing any tooltip for the enabled state for both the kebab menu and the text button.
One doubt: Do we not show the tooltip for both disabled and enabled states or only show the tooltip for the disabled state for the text button? cc @kywalker-rh

Reply:
Good question @manaswinidas. Taking another look at this I would change the tooltip to only being present when the button is disabled, and then I would change the text to:

> To access pipeline server actions, first create a pipeline server.

This prevents the tooltip from getting in the way when performing actions. It also makes it consistent, since we don't have a tooltip for other kebab action menus in the UI.

Reply:
I have added the tooltip for the disabled state for both the kebab menu on the Projects page and the button dropdown on the Global Pipelines page.

Reply:
Looks good! Thanks for updating it.